[IMP] odoo_sh doc update (#1597)
odoo/documentation: content/administration/odoo_sh/advanced/scheduled_actions.rst

.. _odoosh-advanced-scheduled_actions:

=================
Scheduled actions
=================

Special worker
--------------

> On Odoo.sh, cron jobs and incoming mails are handled by a specific
> worker that is not accounted in the workers on your subscription. It is a kind of "hidden free
> worker" bound to every Odoo.sh instance.

Reviewer: - 100th character line break. see our [rst guidelines]([url](https://www.odoo.com/documentation/15.0/contributing/documentation/rst_guidelines.html#start-a-new-line-before-the-100th-character))

```suggestion
On Odoo.sh, cron jobs and incoming mails are handled by a specific worker that is not accounted in
the workers on your subscription. It is a kind of "hidden free worker" bound to every Odoo.sh
instance.
```

By default, this worker can spawn multiple concurrent threads to execute scheduled actions. The
number of threads can be dynamically increased at runtime. However, its exact amount cannot be
predicted as it varies according to the number of extra workers in the subscription, the current
concurrency of incoming mails, the current server's load, etc.

Limitations
-----------

> * The delay between a scheduled task's next execution date and the actual triggering depends on many
>   parameters and can be delayed according to the state of the execution pool of the shared server.
> * Scheduled actions on staging branches are executed less often than production build.
> * The platform will deactivate scheduled actions that repeatedly time out.
> * If the execution time of a scheduled action is business-critical, then we recommend using a
>   dedicated server as the scheduled actions timeout can be modified in the project settings.
> * We cannot guarantee an exact running time for scheduled actions on the Odoo.sh platform.

Reviewer: - not a guideline either but we'd rather use `-` for lists, for consistency of the code
- present tense for general truths.

```suggestion
- The delay between a scheduled task's next execution date and the actual triggering depends on many
  parameters and can be delayed according to the state of the execution pool of the shared server.
- Scheduled actions on staging branches are executed less often than production build.
- The platform deactivates scheduled actions that repeatedly time out.
- If the execution time of a scheduled action is business-critical, then we recommend using a
  dedicated server as the scheduled actions timeout can be modified in the project settings.
- We cannot guarantee an exact running time for scheduled actions on the Odoo.sh platform.
```

> .. _odoosh-advanced-scheduled_actions:

Reviewer: can you apply this as well if possible?
It's not a guideline but it's to remain consistent with all the newly added ref anchors.

=================
Scheduled actions
=================

Special worker
--------------

On Odoo.sh the scheduled actions are handled by a specific worker which is not accounted in the workers on your subscription,
it's a kind of "hidden free worker" bound to every Odoo.sh instance which processes cron jobs and incoming mails.
By default, this worker can spawn multiple concurrent threads for the execution of scheduled actions. The number of threads
can be dynamically increased at runtime according to the amount of extra workers in the subscription and the current concurrency
of incoming mails but also the current server's load, ....
The exact amount of concurrent scheduled actions threads cannot be predicted.

Limitations
-----------

> The delay between a scheduled task's next execution date and the actual triggering depends on a lot of parameters and can be delayed according of the state of the execution pool of the shared server.
> 
> Scheduled action on staging branches are executed less often than production build.
> 
> The platform will deactivate scheduled actions that repeatedly time out.
> 
> If the execution time of the scheduled action is business critical then a dedicated server could be a solution.
> On dedicated server the scheduled actions timeout can be modified the project settings.
> 
> On the Odoo.sh platform, we cannot guarantee an exact running time for scheduled actions.

Reviewer: - list
- phrasing suggestions 
- some English mistakes

```suggestion
- The delay between a scheduled task's next execution date and the actual triggering depends on many
  parameters and can be delayed according to the state of the execution pool of the shared server.
-  Scheduled actions on staging branches are executed less often than production build.
- The platform will deactivate scheduled actions that repeatedly time out.
- If the execution time of a scheduled action is business-critical, then we recommend using a
  dedicated server as the scheduled actions timeout can be modified in the project settings.
- We cannot guarantee an exact running time for scheduled actions on the Odoo.sh platform.
```
